src/Core/System/SalesChannel/Context/SalesChannelContextRestorer.php

```diff
@@ -79,7 +79,12 @@ public function restoreByOrder(string $orderId, Context $context, array $overrid
             $options[SalesChannelContextService::PAYMENT_METHOD_ID] = $paymentMethodId;
         }
 
-        $delivery = $order->getDeliveries() !== null ? $order->getDeliveries()->first() : null;
+        if (!Feature::isActive('v6.8.0.0')) {
+            $delivery = $order->getDeliveries() !== null ? $order->getDeliveries()->first() : null;
+        } else {
+            $delivery = $order->getDeliveries() !== null ? $order->getPrimaryOrderDelivery() : null;
+        }
```

Comment:
Why do you check for deliveries here? There must be deliveries, if we have a primary order delivery, mustn't we? Except, the association is not loaded. If the association is set, then an empty collection will be set.

```suggestion
        if (!Feature::isActive('v6.8.0.0')) {
            $delivery = $order->getDeliveries()?->first();
        } else {
            $delivery = $order->getPrimaryOrderDelivery();
        }
```

Or
```suggestion
        $delivery = $order->getPrimaryOrderDelivery();
        if (!Feature::isActive('v6.8.0.0')) {
            $delivery = $order->getDeliveries()?->first();
        }
```